pkg/infrastructure/services/deployer.go

```diff
@@ -205,6 +205,7 @@ func (s *serviceDeployer) Deploy() (reconcileAfter time.Duration, err error) {
 		}
 		s.generateEventForDeltaResources("Removed", resourceType, delta.Removed)
 	}
+	reconcileAfter, err = configurePrometheus(s.client, s.instance, s.scheme)
```

Comment:
```suggestion
	// In case required resource needs reconciliation
	if reconcileAfter > 0 {
		return
	}
	reconcileAfter, err = configurePrometheus(s.client, s.instance, s.scheme)
```
This will make sure that Prometheus won't overwrite reconciliation period.

Reply:
I that case we could move this check to `createRequiredResources` itself so that if `createRequiredResources` requires reconcilation then it will return from there and don't make a call `getDeployedResources`